lib/cuke_linter/linters/unique_scenario_names_linter.rb

```diff
@@ -0,0 +1,108 @@
+module CukeLinter
+  # A linter that detects non-unique scenario names
+  class UniqueScenarioNamesLinter < Linter
+
+    def initialize
+      super
+      @scenario_names = {}
+    end
+
+    def rule(model)
+      return nil unless valid_model?(model)
+
+      feature_file = model.get_ancestor(:feature_file)
+      return nil if feature_file.nil?
+      
+      file_path = feature_file.path
+
+      case model
+      when CukeModeler::Rule
+        check_rule(model, file_path)
+      when CukeModeler::Outline
+        check_scenario_outline(model, file_path)
+      when CukeModeler::Scenario
+        check_scenario(model, file_path)
+      else
+        nil
+      end
+    end
+
+    private
+
+    def check_rule(model, file_path)
+      problems = []
+      rule_key = "#{file_path}:#{model.name}"
+
+      model.scenarios.each do |scenario|
+        problem = check_scenario(scenario, rule_key)
+        problems << problem if problem
+      end
+
+      model.outlines.each do |outline|
+        problem = check_scenario_outline(outline, rule_key)
+        problems << problem if problem
+      end
+
+      problems.first
+    end
+
+    def check_scenario(model, scope_key)
+      scenario_name = model.name
+      check_duplicate(scenario_name, scope_key)
+      return nil unless duplicate_name?(scenario_name, scope_key)
+      
+      @message = 'Scenario names are not unique'
+      build_problem(model)
+    end
+
+    def check_scenario_outline(model, scope_key)
+      base_name = model.name
+      scenario_names = []
+
+      model.examples.each do |example|
+        header_row = example.rows.first
+        example.rows[1..].each do |data_row|
+          scenario_name = interpolate_name(base_name, header_row, data_row)
+          scenario_names << scenario_name
+        end
+      end
+
+      scenario_names.each do |scenario_name|
+        check_duplicate(scenario_name, scope_key)
+      end
+
+      duplicates = scenario_names.select { |name| duplicate_name?(name, scope_key) }.uniq
+      return nil if duplicates.empty?
+
+      @message = 'Scenario names created by Scenario Outline are not unique'
```

Comment:
Is it possible to put either duplicated name or two places (origin and duplicated)

Reply:
Yep, now test output looks like this:
```
  Scenario name 'Duplicate Name' is not unique. 
    Original name is on line: 10 
    Duplicate is on: 17
    /path/to/file/features/test.feature:17
```

Reply:
Compare please with other outputs. Place them here

Reply:
```
FeatureFileWithMismatchedNameLinter
  Feature file name does not match feature name.
    /path/to/file/features/test.feature
```

```
TestWithNoVerificationStepLinter
  Test does not have a 'Then' step.
    /path/to/file/features/test.feature:5
```

```
ExampleWithoutNameLinter
  Example grouping has no name
    /path/to/file/features/test.feature:10
```

And this current linter gives output like:

```
UniqueScenarioNamesLinter
  Scenario name 'Duplicate Name' is not unique. 
   Original name is on line: 3 
   Duplicate is on: 24
     /path/to/file/features/test.feature:24
  Scenario name created by Scenario Outline 'eating' is not unique. 
   Original name is on line: 5 
   Duplicate is on: 5
     /path/to/file/features/test.feature:5
```

How is better to place both original and duplicate scenarios in linters output?